odxtools/odxlink.py

```diff
@@ -277,24 +277,40 @@ def update(self, new_entries: Dict[OdxLinkId, Any], *, overwrite: bool = True) -
 @overload
 def resolve_snref(target_short_name: str,
                   items: Iterable[OdxNamed],
-                  expected_type: None = None) -> Any:
+                  expected_type: None = None,
+                  *,
+                  lenient: None = None) -> Any:
     """Resolve a short name reference given a sequence of candidate objects"""
     ...
 
 
 @overload
-def resolve_snref(target_short_name: str, items: Iterable[OdxNamed],
-                  expected_type: Type[TNamed]) -> TNamed:
+def resolve_snref(target_short_name: str,
+                  items: Iterable[OdxNamed],
+                  expected_type: Type[TNamed],
+                  *,
+                  lenient: None = None) -> TNamed:
+    ...
+
+
+@overload
+def resolve_snref(target_short_name: str,
+                  items: Iterable[OdxNamed],
+                  expected_type: Type[TNamed],
+                  *,
+                  lenient: bool = True) -> Optional[TNamed]:
     ...
 
 
 def resolve_snref(target_short_name: str,
                   items: Iterable[OdxNamed],
-                  expected_type: Any = None) -> Any:
+                  expected_type: Any = None,
+                  lenient: Optional[bool] = None) -> Any:
```

Comment:
Why lenient is `Optional[bool]` ?

Reply:
because I have not found a way to specialize a function signature depending on an argument's value, only for its type. (Note that this means that mypy uses the incorrect function signature if `resolve_snrefs(..., lenient=False)` is used, but IMO this is pretty much a corner case which also does not do any harm.)